### PR TITLE
Namespace API registration with case insensitive equality check on Name property

### PR DESCRIPTION
**Pull request by melkio**
Crossposting comment from issue: what about adding two namespaces with different names and the same connection string?

@Particular/azure-maintainers ready to review

Connects to #135

### CODE REVIEW

File: src/Tests/Topology/MetaModel/When_configuring_namespace_configurations.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void Should_add_definition_if_it_does_not_exist()
         public void Should_does_not_add_definition_if_exists()
```
Comment (SeanFeldman) [inline src/Tests/Topology/MetaModel/When_configuring_namespace_configurations.cs]: `Should_does_not_add_definition_if_exists()` ==> `Should_not_add_definition_if_exists()`

File: src/Transport/Topology/MetaModel/NamespaceInfo.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public NamespaceInfo(string name, string connectionString)
         public bool Equals(NamespaceInfo other)
         {
             return other != null
-                   && Name.Equals(other.Name)
+                   && Name.Equals(other.Name, StringComparison.InvariantCultureIgnoreCase)
```
Comment (SeanFeldman) [inline src/Transport/Topology/MetaModel/NamespaceInfo.cs]: `StringComparison.InvariantCultureIgnoreCase` should be replaced with `StringComparison.OrdinalIgnoreCase`

File: src/Tests/Topology/MetaModel/When_configuring_namespace_configurations.cs (hunk 1 above)
Comment (melkio) [inline src/Tests/Topology/MetaModel/When_configuring_namespace_configurations.cs]: Ops...sorry :)

Comment (SeanFeldman) [inline src/Tests/Topology/MetaModel/When_configuring_namespace_configurations.cs]: All good  :)

**Comment by melkio** [comment]
@Particular/azure-maintainers still WIP because this comment https://github.com/Particular/NServiceBus.AzureServiceBus/issues/135#issuecomment-191926743  isn't clear to me.

**Comment by yvesgoeleven** [comment]
we should check for duplicate connectionstrings, and not allow them. It would only result in multiple receivers on the same entities, simply a waste of connections

**Comment by melkio** [comment]
@Particular/azure-maintainers ready for review

**Comment by SeanFeldman** [comment]
@melkio when ready, don't forget to remove `[wip]`

**Comment by yvesgoeleven** [comment]
Great work!

**Pull request merged**